pybaseball/datahelpers/postprocessing.py

```diff
@@ -18,24 +18,58 @@
     (re.compile(r'^\d{4}-\d{1,2}-\d{1,2}T\d{2}:\d{2}:\d{2}.\d{1,6}Z$'), '%Y-%m-%dT%H:%M:%S.%fZ'),
 ]
 
+
 def try_parse_dataframe(
     data: pd.DataFrame,
     null_replacement: Union[str, int, float, datetime] = np.nan,
     known_percentages: List[str] = []
 ) -> pd.DataFrame:
-    values = [
-        {column: try_parse(data[column][row_i], column) for column in data.columns}
-        for row_i in range(len(data)) 
-    ]
+    data_copy = data.copy()
+    data_copy = coalesce_nulls(data_copy, null_replacement)
+    data_copy = data_copy.apply(pd.to_numeric, errors='ignore', downcast='integer').convert_dtypes(convert_string=False)
```

Comment:
you're using `convert_types` twice, once here and once just before returning from the function, can we get the same overall effect with just applying `convert_types` just once?

Reply:
@TheCleric this is what I'm talking about, why do we need `to_numeric` and `downcast` and `convert_dtypes`?

Reply:
Mostly because I was trying to keep this function generic for others to use. We can eliminate the numeric parsing or make it optional for statcast to bypass.

Reply:
can we eliminate the downcast? should be unnecessary because `convert_dtypes` should take care of making things ints?